biz.aQute.bndlib/src/aQute/bnd/osgi/Analyzer.java

```diff
@@ -1309,6 +1309,14 @@ private void doNamesection(Jar dot, Manifest manifest) {
 		Instructions instructions = new Instructions(namesection);
 		Set<String> resources = new HashSet<>(dot.getResources()
 			.keySet());
+		// Support package attributes. See
+		// https://docs.oracle.com/javase/8/docs/technotes/guides/versioning/spec/versioning2.html#wp89936
+		dot.getDirectories()
+			.keySet()
+			.stream()
```

Comment:
I think you also need to filter out empty dirs, no?

https://github.com/bndtools/bnd/blob/6c127ce6230ece9bc682e137f5bf90e154e5762a/biz.aQute.bndlib/src/aQute/bnd/osgi/Jar.java#L1018-L1021

It is possible a resource is added which creates a directory and later the resource is removed. This is why getPackages() filters out empty directories.

Reply:
take note that I just added tests and docs.

Reply:
I see your point. I can filter out non-package dirs.

Reply:
fixed